bigbluebutton-html5/imports/api/common/server/helpers.js

```diff
@@ -45,3 +45,20 @@ export const extractCredentials = (credentials) => {
   const requesterUserId = credentialsArray[1];
   return { meetingId, requesterUserId };
 };
+
+export const publicationSafeGuard = function (fn, self) {
+  let stopped = false;
+  const periodicCheck = function () {
+    if (stopped) return;
+    if (!fn()) {
+      self.added(self._name, 'publication-stop-marker', { stopped: true });
+      setTimeout(() => self.stop(), 1000);
```

Comment:
Is this `setTimeout` to call stop still needed?

Reply:
Nope, I implemented a listener to get the stop marker and not depend on setTimout and observer polling

Reply:
Great, thank you.